### PR TITLE
Add .Site.Info.IsServer flag when using built-in Hugo server.

### LINKED ISSUE #4478: Add .Site.IsServer to determine if the site was built in server mode
As per the following Discourse discussion:

https://discourse.gohugo.io/t/disable-portions-of-content-while-developing/10860

It would be helpful to put a condition into a template that can determine whether the page is being built/server by `hugo server` (and not necessarily with the `-D` flag, for which there is already a variable).

This would be helpful for the case where you want to test the site in its final form (no drafts) but want to switch off, e.g. analytics or ads.

### PR DESCRIPTION
**Pull request by felicianotech**
Fixes #4478.

Docs PR coming shortly.

Update:

Docs PR: gohugoio/hugoDocs#430

### CODE REVIEW

File: hugolib/site.go [hunk 1]
```diff
@@ -380,6 +380,7 @@ type SiteInfo struct {
 	BuildDrafts           bool
 	canonifyURLs          bool
 	relativeURLs          bool
+	IsServer              bool
```
Comment (bep) [inline hugolib/site.go]: I think what we want to do is just add a method that returns the value of this{

https://github.com/gohugoio/hugo/blob/3cdf19e9b7e46c57a9bb43ff02199177feb55768/deps/deps.go#L206

Comment (bep) [reply hugolib/site.go]: Or more specifc this value:

Comment (felicianotech) [reply hugolib/site.go]: Ah gotcha. Will rework.

File: docs/content/variables/site.md [hunk 2]
```diff
@@ -122,4 +125,4 @@ You can use `.Site.Params` in a [partial template](/templates/partials/) to call
 <meta name="description" content="{{if .IsHome}}{{ $.Site.Params.description }}{{else}}{{.Description}}{{end}}" />
 {{< /code >}}
 
-[config]: /getting-started/configuration/
\ No newline at end of file
+[config]: /getting-started/configuration/
```
Comment (felicianotech) [inline docs/content/variables/site.md]: I think this is just that Vim added a newline at the end.

**Pull request merged**